lib/SILOptimizer/Transforms/DeadObjectElimination.cpp

```diff
@@ -825,6 +836,42 @@ DeadObjectElimination::removeInstructions(ArrayRef<SILInstruction*> toRemove) {
   }
 }
 
+void DeadObjectElimination::salvageDebugInfo(SILInstruction *alloc,
+                                             SILInstruction *toBeRemoved) {
+  if (auto *SI = dyn_cast<StoreInst>(toBeRemoved)) {
+    auto *parent = SI->getDest()->getDefiningInstruction();
+    if (auto varInfo = buildDIExpression(parent)) {
+      SILBuilderWithScope Builder(SI);
+      Builder.createDebugValue(SI->getLoc(), SI->getSrc(), *varInfo);
+    }
+  }
+}
+
+std::optional<SILDebugVariable>
+DeadObjectElimination::buildDIExpression(SILInstruction *current) {
+  if (!current)
+    return {};
+  if (auto dvci = DebugVarCarryingInst(current)) {
```

Comment:
Nit: LLVM coding style doesn't wrap single statements in `{}`.
```suggestion
  if (auto dvci = DebugVarCarryingInst(current))
    return dvci.getVarInfo();
```